components/datadog/kubernetesagentparams/params.go

```diff
@@ -58,9 +58,23 @@ func NewParams(e config.Env, options ...Option) (*Params, error) {
 	version := &Params{
 		Namespace: defaultAgentNamespace,
 	}
-
 	if e.PipelineID() != "" && e.CommitSHA() != "" {
+		exists, err := e.InternalRegistryImageTagExists(fmt.Sprintf("%s/agent", e.InternalRegistry()), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			return nil, fmt.Errorf("agent image %s not found in the internal registry", fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
+		}
 		options = append(options, WithAgentFullImagePath(utils.BuildDockerImagePath(fmt.Sprintf("%s/agent", e.InternalRegistry()), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))))
+
+		exists, err = e.InternalRegistryImageTagExists(fmt.Sprintf("%s/cluster-agent", e.InternalRegistry()), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			return nil, fmt.Errorf("cluster agent image %s not found in the internal registry", fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
```

Comment:
Could we also add the `e.InternalRegistry()` similarly to what you did in `params.go`?

Reply:
Sure, I will add that